tree/dataframe/inc/ROOT/RDFActionHelpers.hxx

```diff
@@ -323,6 +324,69 @@ public:
    HIST &PartialUpdate(unsigned int slot) { return *fTo->GetAtSlotRaw(slot); }
 };
 
+class FillTGraphHelper : public ROOT::Detail::RDF::RActionImpl<FillTGraphHelper> {
+public:
+   using TGraph = ::TGraph;
+   using Result_t = TGraph;
+
+private:
+   std::shared_ptr<TGraph> fResultGraph;
+   std::unique_ptr<ROOT::TThreadedObject<TGraph>> fTo;
+   bool fIsSortingRequired;
+
+public:
+   FillTGraphHelper(FillTGraphHelper &&) = default;
+   FillTGraphHelper(const FillTGraphHelper &) = delete;
+
+   // The last parameter is always false, as at the moment there is no way to propagate the parameter from the user to this method
+   FillTGraphHelper(const std::shared_ptr<TGraph> &h, const unsigned int nSlots, const bool &isSortingRequired = false)
+      : fResultGraph(h), fTo(new ROOT::TThreadedObject<TGraph>(*h)), fIsSortingRequired(isSortingRequired)
+   {
+      fTo->SetAtSlot(nSlots, h);
+      // Initialise all other slots
+      for (unsigned int i = 0; i < nSlots; ++i) {
+         fTo->GetAtSlot(i);
+      }
+   }
+
+   void Initialize() {}
+   void InitTask(TTreeReader *, unsigned int) {}
+
+   template <typename X0, typename X1,
+             typename std::enable_if<
+                ROOT::TypeTraits::IsContainer<X0>::value && ROOT::TypeTraits::IsContainer<X1>::value, int>::type = 0>
+   void Exec(unsigned int slot, const X0 &x0s, const X1 &x1s)
+   {
+      auto thisSlotH = fTo->GetAtSlotRaw(slot);
+      if (x0s.size() != x1s.size()) {
```

Comment:
This check should be put before anything else. If it is not the case in other helpers, they should be corrected too...

Reply:
https://github.com/root-project/root/blob/cea75adb8435797e1faa9009e0a29558cc1b3c5e/tree/dataframe/inc/ROOT/RDFActionHelpers.hxx#L260